platform/web/serve.py

```diff
@@ -30,7 +31,15 @@ def serve(root, port, run_browser):
     if run_browser:
         # Open the served page in the user's default browser.
         print("Opening the served URL in the default browser (use `--no-browser` or `-n` to disable this).")
-        shell_open(f"http://127.0.0.1:{port}")
+        # Delay opening so that the web server is fully operational
+        # by the time the browser tries to load the page.
+        time.sleep(0.1)
```

Comment:
Maybe I'm missing something, but the webserver is spawned by `test()` at the end of this method, right? So how does `sleep()` *before* that helps?

Reply:
Indeed, this seems to only help in certain circumstances (namely, only if opening the file directly and not relying on displaying a directory index).

I guess we need to find a way to asynchronously wait (without too much boilerplate code if possible). Does modern Python async functionality make this easy enough?

Reply:
> It does help in general still.

as the `sleep()` is added into the `if` block, I assume the goal is this:
```
- run server
- sleep
- open browser
```

but it's currently:
```
- sleep
- open browser
- run server
```
...so it will not improve delay between server start and opening the browser (?)

I'd say async isnt the best tool to handle scenario like this, `threading.Thead` could be better:
```
def shell_open(url, delay=0):
    def inner():
        sleep(delay)
        subprocess.call(["open", url])
    Thread(inner()).start()
```
which will result in:
```
- `shell_open` will start thread
- run server
- (thread wakes up) open browser
```

Reply:
Thanks for providing this code :slightly_smiling_face: 

Unfortunately, this solution doesn't appear to work either. If I put a `print()` statement just after a `shell_open()` call, I don't see the print right away, but logically, I should be able to.

Reply:
sorry, the correct form is:

```diff
def shell_open(url, delay=0):
    def inner():
        sleep(delay)
        subprocess.call(["open", url])
-   Thread(inner()).start()
+   Thread(target=inner).start()
```

Reply:
It works, thanks!